core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -214,7 +216,7 @@ class ReplicaManager(val config: KafkaConfig,
 
   def startup() {
     // start ISR expiration thread
-    scheduler.schedule("isr-expiration", maybeShrinkIsr, period = config.replicaLagTimeMaxMs, unit = TimeUnit.MILLISECONDS)
+    scheduler.schedule("isr-expiration", maybeShrinkIsr, period = config.replicaLagTimeMaxMs/5, unit = TimeUnit.MILLISECONDS)
```

Comment:
Yes, it is indented. I think this makes the guarantee of `replicaLagTimeMaxMs` better. Otherwise, a replica can be stay in ISR for up to `2 x replicaLagTimeMaxMs` even if it is out of sync. Say `replicaLagTimeMaxMs = 100 sec`, the follower's lastCatchUpTime = t1. maybeShrinkIsr() is executed at time t1 + 99 and time t1 + 199. The follower will be able to stay in ISR for 199 seconds and only removed from ISR at time t1 + 199. We can reduce this inaccuracy to 20% by running `maybeShrinkIsr` once every replicaLagTimeMaxMs/5.